api/service/explorer/storage.go

```diff
@@ -97,6 +108,9 @@ func (storage *Storage) Dump(block *types.Block, height uint64) {
 		}
 		storage.UpdateStakingTxAddress(explorerTransaction, tx)
 	}
+
+	// save checkpoint of block dumped
+	storage.GetDB().Put([]byte(DumpCheckpointKey), blockNum.Bytes(), nil)
```

Comment:
if check point is 10, and a block with 20 is dumped. Then the checkpoint is set to 20 and 11-19 blocks won't be dumped anymore. The checkpointing mechanism will be more complicated than this. I would recommend just add a key for each block to mark whether they are dumped or not. It will be much simpler, just a bit more db space needed.

Reply:
good point. I thought that was not possible, but checking the code, during state sync, its actually possible as dump doesn't necessarily dump blocks from genesis in order.